src/lib.rs

```diff
@@ -419,7 +419,9 @@ pub trait IOCached<K, V> {
     }
 
     /// Set the lifespan of cached values, returns the old value
-    fn cache_set_lifespan(&mut self, _seconds: u64) -> Option<u64> {
+    ///
+    /// If the lifespan is set to `None`, cached values do not expire.
```

Comment:
Instead of changing the signature here and having the workaround in the redis impl, I think I'd rather have it be a separate optional trait method `fn cache_unset_lifespan` - and to add it to the standard `Cached` trait also for consistency. The doc can mention only certain types support this and for those that don't it will be a no-op